bootstrapping.py

```diff
@@ -3,391 +3,382 @@
+    # Else...
+    else:
+        # Intialize the vcf file.
+        data = open(g_state['args'].vcf_file, 'rt')
+    all_lines = data.readlines()
+    data.close()
+
     # For every line in the vcf file...
-    for line in data:
+    line_ctr = 0
+    while all_lines[line_ctr].startswith('#'):
+        line = all_lines[line_ctr]
+        line_ctr += 1
         # If the current line is a part of the meta information...
         if line.startswith('##'):
             # Continue to the next line in the vcf file.
             continue
         # Else-if the current line is the header line...
         elif line.startswith('#'):
-            # Intialize the fatal error tracker.
-            fatal = False
             # Split the header line by tabs.
             header = line.split()
             # For every focal population...
-            for key in pop_dicc.keys():
+            for key in g_state['pop_dicc'].keys():
                 # For every sample in the current population...
-                for ind in pop_dicc[key]['IND']:
+                for ind in g_state['pop_dicc'][key]['IND']:
                     # If the sample appears in the header line...
                     if ind in header:
                         # Append the sample's column index w/in the vcf file.
-                        pop_dicc[key]['IDX'].append(header.index(ind))
+                        g_state['pop_dicc'][key]['IDX'].append(header.index(ind))
                     # Else...
                     else:
                         # Remove the sample from the sample list.
-                        pop_dicc[key]['IND'].remove(ind)
+                        g_state['pop_dicc'][key]['IND'].remove(ind)
                         # Compose the error message.
                         err = 'QC: {0} from {1} does not appear in the vcf header...'.format(ind, key)
                         # Log the error.
                         log_file.write(err+'\n')
                 # If the current population has no samples in the vcf file...
-                if len(pop_dicc[key]['IDX']) == 0:
+                if len(g_state['pop_dicc'][key]['IDX']) == 0:
                     # Compose the error message.
                     err = 'FATAL: no {0} samples appear in the vcf header...'.format(key)\
                     +' make sure samples in the meta data file are identical to how they'\
                     +' appear in the vcf header...'
                     # Log the error.
                     log_file.write(err+'\n')
-                    # Record the fatal error.
-                    fatal = True
-            # If a fatal error occured...
-            if fatal:
-                # Stop all calculations.
-                break
-            # Else...
-            else:
-                # Intialize a dictinary to store site patterns by position.
-                site_patterns = {}
-        # Else...
-        else:
-            # Split the header line by tabs.
+                    sys.exit(1)
```

Comment:
What is this wizardy?

Reply:
Since we stop all calculations on error, it seemed better to just exit without any preamble after logging. using `sys.exit(1)` instead of `exit(1)` makes sure it will exit even if multiple threads are running, a non-0 argument is used by convention to indicate that an error occurred.